<a href="https://colab.research.google.com/github/thisaraniNJ/Aircraft_Damage_Prediction/blob/main/Aircraft_Damage_Prediction.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Importing Necessary Libraries

In [11]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from warnings import filterwarnings
filterwarnings('ignore')

pd.set_option('display.max_columns', None) # To represent all the columns in the dataset

# Loading Data

In [12]:
Aviation = pd.read_csv('/content/drive/MyDrive/My ML Projects/AviationData.csv', encoding='mac_roman')

In [13]:
Aviation.head()

Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          Location        Country   Latitude  Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States        NaN        NaN          NaN   
1   BRIDGEPORT, CA  United States        NaN        NaN          NaN   
2    Saltville, VA  United States  36.922223 -81.878056          NaN   
3       EUREKA, CA  United States        NaN        NaN          NaN   
4       Canton, OH  United States        NaN        NaN          NaN   

  Airport.Name Injury.Severity Aircraft.damage Aircraft.Category  \
0          NaN        Fatal(2)       Destroyed               NaN   
1          NaN        Fatal(4)       Destroyed               NaN   
2          NaN        Fatal(3)       Destroyed               NaN   
3          NaN        Fatal(2)       Destroyed               NaN   
4          NaN        Fatal(1)       Destroyed               NaN   

  Registration.Number      Make     Model Amateur.Built  Number.of.Engines  \
0              NC6404   Stinson     108-3            No                1.0   
1              N5069P     Piper  PA24-180            No                1.0   
2              N5142R    Cessna      172M            No                1.0   
3              N1168J  Rockwell       112            No                1.0   
4               N15NY    Cessna       501            No                NaN   

     Engine.Type FAR.Description Schedule Purpose.of.flight Air.carrier  \
0  Reciprocating             NaN      NaN          Personal         NaN   
1  Reciprocating             NaN      NaN          Personal         NaN   
2  Reciprocating             NaN      NaN          Personal         NaN   
3  Reciprocating             NaN      NaN          Personal         NaN   
4            NaN             NaN      NaN          Personal         NaN   

   Total.Fatal.Injuries  Total.Serious.Injuries  Total.Minor.Injuries  \
0                   2.0                     0.0                   0.0   
1                   4.0                     0.0                   0.0   
2                   3.0                     NaN                   NaN   
3                   2.0                     0.0                   0.0   
4                   1.0                     2.0                   NaN   

   Total.Uninjured Weather.Condition Broad.phase.of.flight   Report.Status  \
0              0.0               UNK                Cruise  Probable Cause   
1              0.0               UNK               Unknown  Probable Cause   
2              NaN               IMC                Cruise  Probable Cause   
3              0.0               IMC                Cruise  Probable Cause   
4              0.0               VMC              Approach  Probable Cause   

  Publication.Date  
0              NaN  
1       19-09-1996  
2       26-02-2007  
3       12-09-2000  
4       16-04-1980

In [14]:
Aviation.shape

(87282, 31)

There are 87282 flights have been captured to make a prediction

# Data Description

Non-Null Values, Datatypes, % of null values

In [18]:
info_dict = {
    "Column": Aviation.columns.tolist(),
    "Non-Null Count" : Aviation.count(),
    "Dtypes": Aviation.dtypes.tolist(),
    "Null Precentage":(Aviation.shape[0]-Aviation.count())/Aviation.shape[0]*100
}

Description = pd.DataFrame(info_dict)


In [19]:
Description

Column  Non-Null Count   Dtypes  \
Event.Id                              Event.Id           87282   object   
Investigation.Type          Investigation.Type           87282   object   
Accident.Number                Accident.Number           87282   object   
Event.Date                          Event.Date           87282   object   
Location                              Location           87230   object   
Country                                Country           87056   object   
Latitude                              Latitude           33073   object   
Longitude                            Longitude           33064   object   
Airport.Code                      Airport.Code           49217   object   
Airport.Name                      Airport.Name           51870   object   
Injury.Severity                Injury.Severity           86399   object   
Aircraft.damage                Aircraft.damage           84272   object   
Aircraft.Category            Aircraft.Category           30783   object   
Registration.Number        Registration.Number           85900   object   
Make                                      Make           87219   object   
Model                                    Model           87190   object   
Amateur.Built                    Amateur.Built           87180   object   
Number.of.Engines            Number.of.Engines           81533  float64   
Engine.Type                        Engine.Type           81372   object   
FAR.Description                FAR.Description           30458   object   
Schedule                              Schedule           12330   object   
Purpose.of.flight            Purpose.of.flight           81493   object   
Air.carrier                        Air.carrier           15971   object   
Total.Fatal.Injuries      Total.Fatal.Injuries           75881  float64   
Total.Serious.Injuries  Total.Serious.Injuries           74772  float64   
Total.Minor.Injuries      Total.Minor.Injuries           75349  float64   
Total.Uninjured                Total.Uninjured           81370  float64   
Weather.Condition            Weather.Condition           83304   object   
Broad.phase.of.flight    Broad.phase.of.flight           61724   object   
Report.Status                    Report.Status           82067   object   
Publication.Date              Publication.Date           73629   object   

                        Null Precentage  
Event.Id                       0.000000  
Investigation.Type             0.000000  
Accident.Number                0.000000  
Event.Date                     0.000000  
Location                       0.059577  
Country                        0.258931  
Latitude                      62.107880  
Longitude                     62.118192  
Airport.Code                  43.611512  
Airport.Name                  40.571939  
Injury.Severity                1.011663  
Aircraft.damage                3.448592  
Aircraft.Category             64.731560  
Registration.Number            1.583373  
Make                           0.072180  
Model                          0.105405  
Amateur.Built                  0.116863  
Number.of.Engines              6.586696  
Engine.Type                    6.771156  
FAR.Description               65.103916  
Schedule                      85.873376  
Purpose.of.flight              6.632524  
Air.carrier                   81.701840  
Total.Fatal.Injuries          13.062258  
Total.Serious.Injuries        14.332852  
Total.Minor.Injuries          13.671777  
Total.Uninjured                6.773447  
Weather.Condition              4.557641  
Broad.phase.of.flight         29.282097  
Report.Status                  5.974886  
Publication.Date              15.642400